src/ConcordiumNetSdk/Transactions/TransactionSigner.cs

```diff
@@ -1,31 +1,106 @@
+using ConcordiumNetSdk.Types;
 
 namespace ConcordiumNetSdk.Transactions;
 
-// todo: implement tests
+/// <summary>
+/// Dictionary based implementation of <see cref="ITransactionSigner"/>.
+///
+/// Maintains a dictionary mapping pairs of <see cref="AccountCredentialIndex"/>
+/// and <see cref="AccountKeyIndex"/> to <see cref="ISigner"/>s corresponding
+/// to the sign keys of an account.
+/// </summary>
 public class TransactionSigner : ITransactionSigner
 {
-    private const int MaxCredentialIndexes = 255;
-    private const int MaxKeyIndexes = 255;
-    
-    private readonly Dictionary<Index, Dictionary<Index, ISigner>> _signers = new();
+    /// <summary>
+    /// Internal representation of the signer map.
+    /// </summary>
+    private readonly Dictionary<
+        AccountCredentialIndex,
+        Dictionary<AccountKeyIndex, ISigner>
+    > _signers = new();
 
-    public Dictionary<Index, Dictionary<Index, ISigner>> SignerEntries => _signers;
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TransactionSigner"/> class.
+    /// </summary>
+    public TransactionSigner()
+    {
+        _signers = new Dictionary<AccountCredentialIndex, Dictionary<AccountKeyIndex, ISigner>>();
+    }
 
-    public uint SignatureCount => (uint) _signers.Values.SelectMany(x => x.Values).Count();
+    public byte GetSignatureCount()
+    {
+        return (byte)_signers.Values.SelectMany(x => x.Values).Count();
+    }
 
-    public void AddSignerEntry(Index credentialIndex, Index keyIndex, ISigner signer)
+    /// <summary>
+    /// Gets a dictionary mapping pairs of credential and key indices to the sign key implementation.
+    /// </summary>
+    /// <returns>
+    /// A dictionary mapping pairs of credential and key indices to the sign key implementation.
+    /// </returns>
+    public ImmutableDictionary<
+        AccountCredentialIndex,
+        ImmutableDictionary<AccountKeyIndex, ISigner>
+    > GetSignerEntries()
     {
-        if (_signers.Count > MaxCredentialIndexes) throw new InvalidOperationException($"The maximum length of the credential indexes is {MaxCredentialIndexes}.");
+        return _signers
+            .Select(
+                x =>
+                    new KeyValuePair<
+                        AccountCredentialIndex,
+                        ImmutableDictionary<AccountKeyIndex, ISigner>
+                    >(x.Key, x.Value.ToImmutableDictionary())
+            )
+            .ToImmutableDictionary();
+    }
 
-        if (!_signers.ContainsKey(credentialIndex)) 
+    /// <summary>
+    /// Adds a sign key implementation to the <see cref="TransactionSigner"/>.
+    /// </summary>
+    /// <param name="credentialIndex">The credential index of the credential to which the key belongs.</param>
+    /// <param name="keyIndex">The key index of the key, relative to the credential index.</param>
+    /// <param name="signer">The sign key implementation.</param>
+    public void AddSignerEntry(
```

Comment:
Why is this a public method?

Reply:
Would you rather that a constructor is fed a `Dictionary<AccountCredentialIndex, Dictionary<AccountKeyIndex, ISigner>>`?

Reply:
Is this something users are expected to do? Construct the signer from such a map?

What is the ergonomics of each option?

Reply:
No, but we need to be able to add entries for this signer implementation to be useful. This perhaps allows for a convenient way of constructing a custom signer, however if we only care about importing keys from wallets this class may be redundant altogether since the resulting wallet could just implement `ITransactionSigner`.

Reply:
I suppose we should allow custom cases that are not importing from wallets.